File: src/main/java/com/tom/student/Student.java
package com.tom.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    //IntelliJ??????  alt+Insert
    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }
    public int highest(){
//        int max = (english>math)? english : math;
        return (english > math) ? english : math;
    }
    public int getAverage(){
        return (english+math)/2;
    }
    public void print(){
        int average = getAverage();
        char grading = 'F';
        System.out.print(name + "\t" + english + "\t" + math
        + "\t" + getAverage() + "\t" + ((getAverage()>=60)? "PASS": "FAILED") );

        switch(average/10){
            case 10:
            case 9:
                 grading = 'A';
                 break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
                break;
        }
        System.out.println("\t" + grading);
//        if(getAverage()>=60){
//            System.out.println("\t PASS");
//        } else{
//            System.out.println("FAILED");
//        }

}
}

File: src/main/kotlin/com/tom/kotlin/Student.kt
package com.tom.kotlin

import java.util.*
import kotlin.math.max

fun main() {
//    userinput()
    val stu = Student("Andrei", 77, 99)
    stu.print()
    stu.nameCheck()
    println("Higher score: ${stu.highest()}")

    val test = 123;
    println("Test is $test")

}


class Student(var name: String?, var english: Int, var math: Int){

    fun print(){
//        print(name + "\t" + english + "\t" + math + "\t"
//                + getAverage() + passOrfailed())
//        println("\t" + grading())
        println("$name\t$english\t$math\t${getAverage()}\t${passOrfailed()}\t${grading()}")
    }
    fun getAverage()= (english+math)/2

    fun passOrfailed() = if(getAverage()>=60) "\tPASS" else "FAILED"

    fun grading() = when (getAverage()) {
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    fun highest()=  if(english>math){println("english")
        english} else {println("math")
        math}

    fun nameCheck()= println(name?.length)
    }


fun userinput() {
    val scanner = Scanner(System.`in`)


    print("Please enter your name")
//    var name = scanner.next()
    var name = null
    print("Please enter english score")
    var english = scanner.nextInt();
    print("Please enter math score")
    var math = scanner.nextInt()

    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()


}